docs/reference/docs/delete-by-query.asciidoc

```diff
@@ -0,0 +1,308 @@
+[[docs-delete-by-query]]
+== Delete By Query API
+
+experimental[The delete-by-query API is new and should still be considered experimental.  The API may change in ways that are not backwards compatible]
+
+The simplest usage of `_delete_by_query` just performs a deletion on every
+document in the index. This is useful to delete a subset of documents matching
+a query. Here is the API:
+
+[source,js]
+--------------------------------------------------
+DELETE /twitter/_delete_by_query
+{
+  "query": { <1>
+    "match": {
+      "msg": "some message"
+    }
+  }
+}
+--------------------------------------------------
+// CONSOLE
+// TEST[setup:big_twitter]
+
+<1> The query must be passed as a value to the `query` key, in the same
+way as the <<search-search,Search API>>. You can also use the `q`
+parameter in the same way as the search api.
+
+That will return something like this:
+
+[source,js]
+--------------------------------------------------
+{
+  "took" : 147,
+  "timed_out": false,
+  "deleted": 120,
+  "batches": 2,
+  "version_conflicts": 0,
+  "noops": 0,
+  "retries": 0,
+  "throttled_millis": 0,
+  "requests_per_second": "unlimited",
+  "throttled_until_millis": 0,
+  "total": 120,
+  "failures" : [ ]
+}
+--------------------------------------------------
+// TESTRESPONSE[s/"took" : 147/"took" : "$body.took"/]
+
+`_delete_by_query` gets a snapshot of the index when it starts and deletes what
+it finds using `internal` versioning. That means that you'll get a version
+conflict if the document changes between the time when the snapshot was taken
+and when the delete request is processed. When the versions match the document
+is deleted.
+
+All delete and query failures cause the `_delete_by_query` to abort and are
```

Comment:
I think this is a part of update-by-query and reindex and delete-by-query's docs that'll have to be changed. Rejected executions are retried for bulk, and, with #18331, will be for searches.